tags/views.py

```diff
@@ -0,0 +1,115 @@
+from django.http import HttpResponseBadRequest, HttpResponse, JsonResponse
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework import mixins
+from rest_framework import viewsets
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.views import APIView
+from rest_framework.viewsets import GenericViewSet
+
+from tags.models import Namespace, Tag
+from tags.permissions import (
+    NamespacePermission,
+    user_can_link_tag_to,
+    ManageTagPermission,
+)
+from tags.serializers import NamespaceSerializer, TagSerializer
+
+"""
+
+To create a namespace : 
+- either be an admin and create a global scope
+- either specify a scope and a scoped_to and have permissions on the scoped_to object
+
+"""
+
+
+class NamespaceViewSet(viewsets.ModelViewSet):
+    queryset = Namespace.objects.all()
+    serializer_class = NamespaceSerializer
+    filter_backends = (DjangoFilterBackend,)
+    filterset_fields = ("scope", "scoped_to")
+    permission_classes = (NamespacePermission,)
+
+    def create(self, request, *args, **kwargs):
+        # Required since we use "unique_together"
+        if "scoped_to" not in request.data:
+            request.data["scoped_to"] = ""
```

Comment:
Er… how can a Namespace be `scoped_to` nothing actually?

Reply:
Yes, if the scope is global